src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -157,7 +163,299 @@ public SPDX30Parser(
+                    case string name when name.Contains("Package"):
+                        elementsResult.PackagesCount += 1;
+                        break;
+                    case string name when name.Contains("ExternalMap"):
+                        elementsResult.ReferencesCount += 1;
+                        break;
+                    case string name when name.Contains("Relationship"):
+                        elementsResult.RelationshipsCount += 1;
+                        break;
+                    default:
+                        Console.WriteLine($"Unrecognized entity type: {entityType?.Name}");
+                        break;
+                }
+            }
+        }
+
+        // Validate if elements meet required compliance standards
+        switch (complianceStandardAsEnum)
+        {
+            case ComplianceStandard.NTIA:
+                ValidateNTIARequirements(elementsList);
+                break;
+        }
+
+        elementsResult.Elements = elementsList;
+        elementsResult.Files = filesList;
+
+        return elementsResult;
+    }
+
+    public Type GetEntityType(JsonObject jsonObject, ComplianceStandard? requiredComplianceStandard)
+    {
+        var assembly = typeof(Element).Assembly;
+        var typeFromSbom = jsonObject["type"]?.ToString();
+        var entityType = typeFromSbom;
+
+        // For these special cases, remove the prefix from the type.
+        switch (typeFromSbom)
+        {
+            case "software_File":
+                entityType = "File";
+                break;
+            case "software_Package":
+                entityType = "Package";
+                break;
+        }
+
+        // If the entity type is in the list of entities that require different NTIA requirements, then add the NTIA prefix.
+        switch (requiredComplianceStandard)
+        {
+            case ComplianceStandard.NTIA:
+                if (this.EntitiesToEnforceComplianceStandardsFor?.Contains(entityType) == true)
+                {
+                    entityType = "NTIA" + entityType;
+                }
+
+                break;
+        }
+
+        var type = assembly.GetType($"Microsoft.Sbom.Parsers.Spdx30SbomParser.Entities.{entityType}") ?? throw new ParserException($"Type \"{typeFromSbom} on {jsonObject} is invalid.");
+
+        return type;
+    }
+
+    public void ValidateNTIARequirements(List<Element> elementsList)
+    {
+        ValidateSbomDocCreationForNTIA(elementsList);
+        ValidateSbomFilesForNTIA(elementsList);
+        ValidateSbomPackagesForNTIA(elementsList);
+    }
+
+    /// <summary>
+    /// Validate that information about the SBOM document is present.
+    /// </summary>
+    /// <param name="elementsList"></param>
+    /// <exception cref="ParserException"></exception>
+    public void ValidateSbomDocCreationForNTIA(List<Element> elementsList)
+    {
+        var spdxDocumentElements = elementsList.Where(element => element is SpdxDocument);
```

Comment:
`Where` returns an `IEnumerable`, which has some benefits, but the first thing that happens is to call `Count()`, which walks the entire list. Would the code to simpler if you made `spdxDocumentElements` a `List` by appending `.ToList()` on line 339?